src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -658,14 +658,22 @@ private unsafe int HandleEventShutdownInitiatedByTransport(ref SHUTDOWN_INITIATE
     {
         Exception exception = ExceptionDispatchInfo.SetCurrentStackTrace(ThrowHelper.GetExceptionForMsQuicStatus(data.Status, (long)data.ErrorCode));
         _connectedTcs.TrySetException(exception);
-        _connectionCloseTcs.TrySetException(exception);
+        if (_connectionCloseTcs.TrySetException(exception))
```

Comment:
Should we extract it as a method? as it's used 3 times

Reply:
I removed it from `ThrowHelper` as it's only 2 liner and only used for `_connectionCloseTcs`. But I can shuffle it back, I don't care that much about it.